### PR TITLE
Extract mule functionality to individual functions

### LINKED ISSUE #28: Handling ancillary files - flag as error
Ancillary files are the same UM fieldsfile format but don't have some of the header information of model output files.

Using `~access/umdir/ancil/atmos/GC5/n96e/orca1/land_sea_mask/etop01/v1/qrparm.mask` fails with the less than useful

`AttributeError: 'NoneType' object has no attribute 'zsea_at_rho'`

Something like
```
    ff = mule.load_umfile(str(infile))
    if isinstance(ff, mule.ancil.AncilFile):
        raise Exception('Ancillary files are not supported')
```

### LINKED ISSUE #29: Extract & test mule functionality
This issue is for mule related testing/refactoring (subset of #14).

Tasks:
* [X] Move mule data extraction to functions
* [X] Grid strings to constants
* [X] #28
* [X] Add basic unit tests

### PR DESCRIPTION
**Pull request by truth-quark**
Fixes #29. Also fixes #28.

Here's the start of retro-fitting unit tests & refactoring `um2nc-standalone`.

Changes include:
* Renaming `src` to the `umpost` root package name
* Extract mule data operations to helper files
* Shrinking `process()` down to a high level workflow func
* Adding some basic unit tests (some are contrived)

Some unit tests are a bit contrived using "empty" fields & ancil files, combined with `mock` to attach data values. I'd like to create synthetic `mule` objects from real data for future testing, but I'll defer this for a future PR, when the code structure is more modular.

For the review, any opinions on the restructure, gaps & correctness checks are welcome...

### CODE REVIEW

File: umpost/um2netcdf.py [hunk 1]
```diff
@@ -16,6 +16,15 @@
 from iris.fileformats.pp import PPField
 
 
+GRID_END_GAME = 'EG'
+GRID_NEW_DYNAMICS = 'ND'
+
+
+class UMError(Exception):
```
[the remark this answers is not in the document]
Comment (truth-quark) [reply umpost/um2netcdf.py:23]: Yep, the current name is misleading, partially from avoiding IDE errors around capitalisation.

How does `PostprocessingError` sound? Too long? Maybe `PostError`?

File: umpost/um2netcdf.py [hunk 2]
```diff
@@ -427,6 +433,65 @@ def process(infile, outfile, args):
...
+    Parameters
+    ----------
+    ff : an open fields file.
+
+    Returns
+    -------
+    (z_rho, z_theta) tuple.
+    """
+    z_rho = ff.level_dependent_constants.zsea_at_rho
+    z_theta = ff.level_dependent_constants.zsea_at_theta
```
[the remark this answers is not in the document]
Comment (truth-quark) [reply umpost/um2netcdf.py:491]: I'm not entirely sure, so have worked on the assumption ancil files lack the `z_sea_<thing>` attributes. There's a ancil file guard in this commit: https://github.com/ACCESS-NRI/um2nc-standalone/pull/31/commits/0445264a3d7bb626d7e2ca9ba02f8a3cd6e60a41, so we have some protection.

As this is a work in progress, I don't mind gaps, as long as the code fails fast & can be improved iteratively.

File: test/test_um2netcdf.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    ff = mule.ff.FieldsFile()
+    ff.fixed_length_header.grid_staggering = 3
+    grid_type = um2nc.get_grid_type(ff)
+    assert grid_type == um2nc.GRID_NEW_DYNAMICS
+
+
+def test_get_grid_type_error():
+    ff = mule.ff.FieldsFile()  # "empty" fields file has no grid staggering
+
+    with pytest.raises(um2nc.UMError):
```
Comment (jo-basevi) [inline test/test_um2netcdf.py:26]: Can also use `match` to check the error message, e.g:
```
with pytest.raises(um2nc.UMError, match="Unable to determine grid staggering from header '{no_staggering_value}'"):
```

File: umpost/um2netcdf.py [hunk 4]
```diff
@@ -427,6 +433,65 @@ def process(infile, outfile, args):
...
+    """
+    Helper function for accessing grid spacing variables.
+
+    Parameters
+    ----------
+    ff : an open fields file.
+
+    Returns
+    -------
+    (row_spacing, column spacing tuple)
```
Comment (jo-basevi) [inline umpost/um2netcdf.py:468]:
```suggestion
    (row_spacing, column_spacing) tuple
```
Some type information could useful but it is also not super necessary.

**Review by jo-basevi — APPROVED** [review]
Changes look good to me! The doc-strings were helpful for me reviewing it

File: umpost/um2netcdf.py [hunk 5]
```diff
@@ -427,6 +433,65 @@ def process(infile, outfile, args):
...
+    """
+    return ff.real_constants.row_spacing, ff.real_constants.col_spacing
```
[the remark this answers is not in the document]
Comment (truth-quark) [reply umpost/um2netcdf.py]: I've updated this to break explicitly with a `NotImplementedError`.

File: umpost/um2netcdf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (truth-quark) [reply umpost/um2netcdf.py]: Also updated this one to break explicitly with a `NotImplementedError`.

File: umpost/um2netcdf.py (hunk 4 above)
Comment (truth-quark) [reply umpost/um2netcdf.py]: Fixed the comment & added datatypes to description.

File: umpost/um2netcdf.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (blimlim) [reply umpost/um2netcdf.py]: Great, sounds good!

File: umpost/um2netcdf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (blimlim) [reply umpost/um2netcdf.py]: That sounds good!

File: umpost/um2netcdf.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (blimlim) [reply umpost/um2netcdf.py:35]: I think either of those should be ok!

[the remark this answers is not in the document]
Comment (truth-quark) [reply umpost/um2netcdf.py]: I've renamed it to `PostProcessingError`.

File: test/test_um2netcdf.py (hunk 3 above)
Comment (truth-quark) [reply test/test_um2netcdf.py]: That's a useful feature. I'll skip testing the error messages until usage requirements are clearer.

**Review by blimlim — APPROVED** [review]
Looks great!

**Comment by truth-quark** [comment]
Nice, thanks all for the reviews!

**Pull request merged**